pkg/inst/worker/daemon.R

```diff
@@ -0,0 +1,70 @@
+# Worker daemon
+
+rLibDir <- Sys.getenv("SPARKR_RLIBDIR")
+script <- paste(rLibDir, "SparkR/worker/worker.R", sep="/")
+
+# preload SparkR package, speedup worker
+.libPaths(c(rLibDir, .libPaths()))
+suppressPackageStartupMessages(library(SparkR))
+
+port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
+
+inputCon <- socketConnection(host = "localhost",
+                             port = port,
+                             blocking = TRUE,
+                             open = "rb",
+                             timeout = 6000,
+                             server = FALSE)
+
+# Read from connection, retrying to read exactly 'size' bytes
+readWithRetry <- function(con, size) {
+  data <- readBin(con, raw(), size, endian = "big")
+  # Since this function is called after
+  # calling `select`, 0 bytes read implies socket is closed
+  if (length(data) == 0) {
+    return(data)
+  }
+  while (length(data) < size) {
```

Comment:
Yeah this is a good point. I originally had a `checkWrite` function where if we read 0 bytes, I'd try to write a few bytes to the socket and add `tryCatch` to return false if the write fails.

I was hoping that we'd not run into this as we always write 4 bytes on the stream but yes we should check for a closed socket if we repeatedly get zero bytes